### DISCUSSION TITLE
Preloading page clears FlashBag

### DISCUSSION THREAD

**Discussion by Moskito89** [symfony/symfony#50682] [Q&A]
Hallo everyone! I found a problem with flash messages when preloading a page: 
If a page is loaded with `<link rel="prefetch" href="url">`, Symfony already delivers the flash messages (which is basically correct). However, they are missing when the page is actually called up. 
Has anyone encountered this before? Is Symfony able to detect a preload request?

**Comment by lyrixx** [comment]
Is there a specific header when a page preloads?

And even if you are able to detect a preload, if the user clicks on the link that leads to the preloaded page, the flash won't be displayed. So the flash should be displayed on the first non-preloaded page. This could be strange for the user isn't?

**Comment by Moskito89** [comment]
There is a header, but they differ from browser to browser: Chrome sends `Purpose: prefetch` and Firefox `X-Moz: prefetch`. Safari can't preload currently...

I think whether the browser caches the HTML or not depends in the end on the settings of the response header. This means: 
1. the response for a preload request contains Flash messages in any case, because it is the first call.
2. whether the browser calls the response later or not depends on the response header.
3. since such a page contains Flash messages, the response cannot be hard-cached. The response header will therefore be configured in such a way that the page will not be cached and the browser will therefore reject the response of the preload.

So the question remains: can we detect preloads in Symfony and prevent Flash messages from being delivered?

**Reply by lyrixx** [reply]
Sure, you look at the header :)

**Reply by Moskito89** [reply]
Sure! 😄 

But surely I can't be the only one struggling with this? Is there no "bigger" solution for this?

**Reply by lyrixx** [reply]
I'm not aware of it, but may be
